### PR TITLE
Improve Batch Renormalization tests

### PR DESCRIPTION
**Pull request by toslunar**
Replace the backward tests of Batch Renormalization by comparisons with its naive implementation, which uses several `FunctionNode`s.

Batch Renormalization uses "partial gradient". The current tests in Chainer use `check_backward` so that they check gradients w.r.t. a modified forward computation.  This PR removes `freeze_running_statistics` option, which has been introduced just for these tests.

Merge #4869 first.

### CODE REVIEW

**Comment by toslunar** [comment]
#4869 is merged.

**Comment by okuta** [comment]
Please rebase this.

**Comment by toslunar** [comment]
Jenkins, test this please.

File: chainer/links/normalization/batch_renormalization.py [hunk 1]
```diff
@@ -35,15 +35,14 @@ class BatchRenormalization(BatchNormalization):
     def __init__(self, size, rmax=1, dmax=0, decay=0.9, eps=2e-5,
                  dtype=numpy.float32, use_gamma=True, use_beta=True,
                  initial_gamma=None, initial_beta=None,
-                 freeze_running_statistics=False):
+                 ):
```
Comment (okuta) [inline chainer/links/normalization/batch_renormalization.py]: You can merge above line.

File: tests/chainer_tests/functions_tests/normalization_tests/test_batch_renormalization.py [hunk 2]
```diff
@@ -19,6 +19,36 @@ def _batch_renormalization(expander, gamma, beta, x, mean, var, r, d):
...
 
+# naive implementation of differentiable batch renormalization
+def _naive_batch_renormalization(
+        x, gamma, beta, rmax, dmax, eps, avg_mean, avg_std, axis):
+    shape = x.shape
+    stat_shape = list(shape)
+    for i in axis:
+        stat_shape[i] = 1
+    stat_shape = tuple(stat_shape)
+    gamma = chainer.functions.broadcast_to(
```
Comment (okuta) [inline tests/chainer_tests/functions_tests/normalization_tests/test_batch_renormalization.py]: You can use broadcast.

File: tests/chainer_tests/functions_tests/normalization_tests/test_batch_renormalization.py [hunk 3]
```diff
@@ -19,6 +19,36 @@ def _batch_renormalization(expander, gamma, beta, x, mean, var, r, d):
...
+        chainer.functions.reshape(gamma, stat_shape), shape)
+    beta = chainer.functions.broadcast_to(
```
Comment (okuta) [inline tests/chainer_tests/functions_tests/normalization_tests/test_batch_renormalization.py]: ditto

File: tests/chainer_tests/functions_tests/normalization_tests/test_batch_renormalization.py [hunk 4]
```diff
@@ -19,6 +19,36 @@ def _batch_renormalization(expander, gamma, beta, x, mean, var, r, d):
...
+        chainer.functions.reshape(beta, stat_shape), shape)
+    avg_mean = avg_mean.reshape(stat_shape)
+    avg_std = avg_std.reshape(stat_shape)
+
+    mean = chainer.functions.broadcast_to(
```
Comment (okuta) [inline tests/chainer_tests/functions_tests/normalization_tests/test_batch_renormalization.py]: ditto.

**Pull request merged**